sklearn/naive_bayes.py

```diff
@@ -1528,3 +1541,600 @@ def _joint_log_likelihood(self, X):
+            self._check_feature_names(X, reset=False)
+            self._check_n_features(X, reset=False)
+        # Consistency checks for X, y are delegated to subestimators
+
+        # Subestimators get original sample_weight. This is for class counts:
+        if sample_weight is not None:
+            weights = _check_sample_weight(sample_weight, X=y, copy=True)
+
+        # Subestimators should've checked classes. We set classes_ for counts
+        # and so that first_call becomes False at next partial_fit call.
+        _check_partial_fit_first_call(self, classes)
+
+        # We don't use sklearn.utils.multiclass.class_distribution, because it
+        # neither returns class_count, nor is suitable for partial_fit.
+        if sample_weight is None:
+            counts = np.zeros(len(self.classes_))
+            for i, c in enumerate(self.classes_):
+                counts[i] = (column_or_1d(y) == c).sum()
+        else:
+            counts = np.zeros(len(self.classes_))
+            for i, c in enumerate(self.classes_):
+                counts[i] = (weights * (column_or_1d(y) == c)).sum()
+
+        if first_call:
+            self.n_classes_ = len(self.classes_)
+            self.class_count_ = counts
+        else:
+            self.class_count_ += counts
+
+        estimators = list(self._iter(fitted=not first_call, replace_strings=True))
+        fitted_estimators = Parallel(n_jobs=self.n_jobs)(
+            delayed(_partial_fit_one)(
+                estimator=clone(nb_estimator) if first_call else nb_estimator,
+                X=_safe_indexing(X, cols, axis=1),
+                y=y,
+                message_clsname="ColumnwiseNB",
+                message=self._log_message(name, idx, len(estimators)),
+                classes=classes,
+                sample_weight=sample_weight,
+            )
+            for idx, (name, nb_estimator, cols) in enumerate(estimators, 1)
+        )
+        self._update_fitted_estimators(fitted_estimators)
+        self._update_class_prior()
+        return self
+
+    @property
+    def _estimators(self):
+        """Internal list of subestimators.
+
+        This is for the implementation of get_params via BaseComposition._get_params,
+        which expects lists of tuples of len 2.
+        """
+        # Implemented in the image and likeness of ColumnTranformer._transformers
+        try:
+            return [(name, e) for name, e, _ in self.nb_estimators]
+        except (TypeError, ValueError):
+            # This try-except clause is needed to pass the test from test_common.py:
+            # test_estimators_do_not_raise_errors_in_init_or_set_params().
+            # ColumnTransformer does the same. See PR #21355 for details.
+            return self.nb_estimators
+
+    @_estimators.setter
+    def _estimators(self, value):
+        # Implemented in the image and likeness of ColumnTranformer._transformers
+        # TODO: Is renaming or changing the order legal? Swap `name` and `_`?
```

Comment:
Remove. You can always swap the name since since you either call `fit` or make the consistancy check in `partial_fit` and `predict` that will raise an error.

Reply:
Removed. I did not realise at the time that swapping `name` and `_` won't have any effect.